Document/0x07-Security-Testing-SDLC.md

```diff
@@ -86,25 +88,42 @@ The following diagram provides an overview of the way to perform test in the SDL
 - no new feature has unvoluntarily been introduced that could endanger the system or its users.
+
 As a consequence, performing one type of tests is not enough to pretend having covered both types and that the produced software is both usable and secure. The same care should be given to both types of tests as they are of the same importance and that final users now put a strong emphasis both on quality (e.g. the fact features that are brought to them perform the way they expect them to) and security (e.g. that they can trust the software vendor that their money will not be stolen or their private life will remain private). 
 
+#### Defining a Test Strategy
+
+The purpose of a test strategy is to define which tests will be performed all along the SDLC and how often. Its goal is twofold: make sure security objectives are met by the final software product, which are generally expressed by customers / legal / marketing / Corporate teams, while bringing cost-effectiveness. The test strategy is generally created at the beginning of a project, after risks have been clarified (Initiation phase) but before code production (Coding phase) starts. It generally takes place during the Architecture and Design phase. It takes inputs from activities such as Risk Management, Threat Modeling, Security Engineering, ...
+
+-- TODO : add diagram (in the form of a workflow) showing inputs of a Test Strategy, and outputs (test cases, ...) --
+
+A Test Strategy does not always need to be formally written: it may be described through Stories (in Agile projects), quickly written in the form of checklists, Test Cases could be written in a given tool, ... ; however, it definitely needs to be shared, as it may be defined by the Architecture team, but will have to be implemented by other teams such as Development, Testing, Quality, ... Moreover, it needs to be agreed upon by all technical teams as it should not place unacceptable burdens on any of them. 
+
+Ideally, a Test Strategy addresses topics such as:
+- objectives to be met and description of risks to be put under control.
+- how these objectives will be met and risks reduced to an acceptable level: which tests will be mandatory, who will perform them, how, when, at which frequency.
+- acceptance criteria of the current project.
+
+In order to follow its effectiveness and progress, metrics should be defined, updated all along the project and periodically communicated. An entire book could be written on the relevant metrics to choose; the best that can be said is that they depend on risk profiles, projects and organizations. However, some examples can be given : metrics include 
+- the number of stories related to security controls that are implemented,
+- code coverage for unit tests on security controls and sensitive features,
+- the number of security bugs found by static analysis tools upon each build,
+- the trend of the backlog for security bugs (may be sorted by criticity).

```

Comment:
Is criticity used in the IT security context? Never heard it. Otherwise maybe use just criticality.